src/rust/src/conversion/mod.rs

```diff
@@ -378,3 +399,38 @@ impl TryFrom<&str> for Wrap<WindowMapping> {
         Ok(Wrap(parsed))
     }
 }
+
+impl TryFrom<&str> for Wrap<SetOperation> {
+    type Error = String;
+
+    fn try_from(operation: &str) -> Result<Self, String> {
+        let parsed = match operation {
+            "union" => SetOperation::Union,
+            "intersection" => SetOperation::Intersection,
+            "difference" => SetOperation::Difference,
+            "symmetric_difference" => SetOperation::SymmetricDifference,
+            v => {
+                return Err(format!(
+                    "`operation` must be one of 'union', 'intersection', 'difference', 'symmetric_difference'. Got '{v}'",
```

Comment:
Since `arg_match0` is used privately on the R side these days, the message here can be simplified.
Something like `unsupported value: '{v}'`